### PR TITLE
Remove insert_record and insert_list

### LINKED ISSUE #1931: Remove `insert_record` and `insert_list` helper functions
### Description

VAST's `record` is internally modeled as a stable map, i.e., a `std::vector` containing pairs of key and corresponding value.

We have helper functions `insert_record` and `insert_list` that add a field to a record and return a reference to it.

```cpp
auto x = record{};
auto& y = insert_record(x, "y");
y["z"] = 42;
// yields the structure x = {y: {z: 42}}
```

However, this invites the user to forget that iterators into stable maps may be invalidated when appending to a record. This code is undefined behavior:

```cpp
auto x = record{};
auto& y = insert_record(x, "y");
y["z"] = 42;
x["a"] = 1337;
// yields the structure x = {y: {z: 42}, a: 1337}, or crashes.
```

Instead, we should rewrite the code like this, which is safe:

```cpp
auto x = record{};
auto y = record{};
y["z"] = 42;
x["y"] = std::move(y);
x["a"] = 1337;
// yields the structure x = {y: {z: 42}, a: 1337}.
```

### Definition of Done

The functions `insert_list` and `insert_record` no longer exist, and code samples are rewritten as shown above.

### Anything else?

See also: tenzir/vast#1915, which was caused by this very issue.

### PR DESCRIPTION
**Pull request by 6yozo**
Fixes #1931 

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

file-by-file.

### CODE REVIEW

File: libvast/src/system/index.cpp [hunk 1]
```diff
@@ -477,35 +477,40 @@ index_state::status(status_verbosity v) const {
...
     worker_status["busy"] = workers - idle_workers.size();
-    auto& pending_status = insert_list(rs->content, "pending");
+    rs->content["workers"] = std::move(worker_status);
+    auto pending_status = list{};
     for (auto& [u, qs] : pending) {
-      auto& q = insert_record(pending_status);
+      auto q = record{};
       q["id"] = fmt::to_string(u);
       q["query"] = fmt::to_string(qs);
+      pending_status.emplace_back(record{}); //CHECK!
```
Comment (dominiklohmann) [inline libvast/src/system/index.cpp:504]:
```suggestion
      pending_status.emplace_back(std::move(q));
```

**Review by dominiklohmann — APPROVED** [review]
It's your first contribution! Welcome 🎉

There's only one minor issue to fix with the change, LGTM aside from that 🚀

File: libvast/src/system/posix_filesystem.cpp [hunk 2]
```diff
@@ -111,13 +111,15 @@ posix_filesystem(filesystem_actor::stateful_pointer<posix_filesystem_state> self
...
+        auto ops = record{};
         auto add_stats = [&](auto& name, auto& stats) {
-          auto& dict = insert_record(ops, name);
+          auto dict = record{};
           dict["successful"] = count{stats.successful};
           dict["failed"] = count{stats.failed};
           dict["bytes"] = count{stats.bytes};
+          ops[name] = std::move(dict);
         };
+        result["operations"] = std::move(ops);
```
Comment (dominiklohmann) [inline libvast/src/system/posix_filesystem.cpp:122]: The add_stats lambda writes to `ops`, which fails after you moved it here.

**Comment by tobim** [comment]
There are a few places where the problematic pattern is used directly, can you also refactor those in the same way?
```
libvast/src/system/active_partition.cpp:517:        auto& ps = caf::get<record>(indexer_states.emplace_back(record{}));
libvast/src/system/index.cpp:519:              auto& ps = caf::get<record>(xs.emplace_back(record{}));
libvast/src/system/index.cpp:532:              auto& ps = caf::get<record>(xs.emplace_back(record{}));
libvast/src/system/type_registry.cpp:52:          = caf::get<record>(concepts_status.emplace_back(record{}));
```

**Pull request merged**